allennlp/sanity_checks/task_checklists/task_suite.py

```diff
@@ -0,0 +1,293 @@
+    * Logic : Ability to handle symmetry, consistency, and conjunctions.
+    * Fairness
+
+
+    # Parameters
+
+    suite: `checklist.test_suite.TestSuite`, optional (default = `None`)
+        Pass in an existing test suite.
+
+    add_default_tests: `bool` (default = `False`)
+        Whether to add default checklist tests for the task.
+
+    data: `List[Any]`, optional (default = `None`)
+        If the data is provided, and `add_default_tests` is `True`,
+        tests that perturb the data are also added.
+
+        For instance, if the task is sentiment analysis, and the a
+        list of sentences is passed, it will add tests that check
+        a model's robustness to typos, etc.
+    """
+
+    _capabilities = [
+        "Vocabulary",
+        "Taxonomy",
+        "Robustness",
+        "NER",
+        "Fairness",
+        "Temporal",
+        "Negation",
+        "Coref",
+        "SRL",
+        "Logic",
+    ]
+
+    def __init__(
+        self,
+        suite: Optional[TestSuite] = None,
+        add_default_tests: bool = False,
+        data: Optional[List[Any]] = None,
+        **kwargs,
+    ):
+        self.suite = suite or TestSuite()
+
+        if add_default_tests:
+            self._default_tests(data, **kwargs)
+
+    def _prediction_and_confidence_scores(self, predictor: Predictor) -> Callable:
+        """
+        This makes certain assumptions about the task predictor
+        input and output expectations. This should return a function
+        that takes the data as input, passes it to the predictor,
+        and returns predictions and confidences.
+        """
+        return NotImplementedError
+
+    def describe(self):
+        """
+        Gives a description of the test suite.
+        """
+
+        def cap_order(x):
+            return self._capabilities.index(x) if x in self._capabilities else 100
+
+        capabilities = sorted(
+            set([x["capability"] for x in self.suite.info.values()]), key=cap_order
+        )
+        print(
+            "\n\nThis suite contains {} tests across {} capabilities.".format(
+                len(self.suite.tests), len(capabilities)
+            )
+        )
+        print()
+        for capability in capabilities:
+            tests = [
+                name for name, test in self.suite.info.items() if test["capability"] == capability
+            ]
+            if len(tests) > 0:
+                print("\n\t{} ({} tests)\n".format(capability, len(tests)))
+                for test in tests:
+                    description = self.suite.info[test]["description"]
+                    num_test_cases = len(self.suite.tests[test].data)
+                    about_test = "\t * {} ({} test cases)".format(test, num_test_cases)
+                    if description:
+                        about_test += " : {}".format(description)
+                    print(about_test)
+
+    def summary(self, capabilities=None, file=sys.stdout, **kwargs):
```

Comment:
`describe()` prints to stdout, but this takes a parameter for the output.

Reply:
Yes, I intended the `describe` function as a tool to examine the suite and the tests it contains (which can be helpful when someone passes in an already created checklist suite as input). The summary will print the results of the test cases comprehensively by actually running the tests.